src/solvers/CMakeLists.txt

```diff
@@ -98,6 +98,36 @@ elseif("${sat_impl}" STREQUAL "glucose")
     target_sources(solvers PRIVATE ${glucose_source})
 
     target_link_libraries(solvers glucose-condensed)
+elseif("${sat_impl}" STREQUAL "cadical")
+    message(STATUS "Building solvers with cadical")
+
+    download_project(PROJ cadical
+        URL https://github.com/arminbiere/cadical/archive/rel-1.3.0.tar.gz
+        PATCH_COMMAND true
+        COMMAND CXX=${CMAKE_CXX_COMPILER} ./configure -O3 -s -j
+        URL_MD5 5bd15d1e198d2e904a8af8b7873dd341
+    )
+
+    message(STATUS "Building CaDiCaL")
+    execute_process(COMMAND make WORKING_DIRECTORY ${cadical_SOURCE_DIR})
+
+    target_compile_definitions(solvers PUBLIC
+        SATCHECK_CADICAL HAVE_CADICAL
+    )
+
+    add_library(cadical STATIC IMPORTED)
+
+    set_target_properties(
+        cadical
+        PROPERTIES IMPORTED_LOCATION ${cadical_SOURCE_DIR}/build/libcadical.a
```

Comment:
This will only work on Linux, is that OK? If cadical also works on Windows/OSX it'd be nice if just used something like `find_library` here to get a more portable library path.

Reply:
Cadical's makefile hard-codes this, so I don't think it'll work out of the box on other platforms.